src/precice/tests/SerialTests.cpp

```diff
@@ -587,6 +587,152 @@ BOOST_AUTO_TEST_CASE(testExplicitWithDataInitialization)
+    // expected data value received
+    BOOST_TEST(3.0 == readDataB[0]);
+    // change value at read destination
+    readDataB[0] = -0.2;
+    BOOST_TEST(-0.2 == readDataB[0]);
+    cplInterface.readScalarData(dataBID, vertexIDs[0], readDataB[0]);
+    // expected data value received
+    BOOST_TEST(3.0 == readDataB[0]);
+
+    // Vector Data
+    cplInterface.readBlockVectorData(dataAID, 1, vertexIDs.data(), readDataA.data());
+    // expected data value received
+    BOOST_TEST(Vector3d(3.1, 3.2, 3.3) == readDataA);
+    // change value at read destination
+    readDataA[0] = -0.2;
+    readDataA[1] = -0.2;
+    readDataA[2] = -0.2;
+    BOOST_TEST(Vector3d(-0.2, -0.2, -0.2) == readDataA);
+    cplInterface.readVectorData(dataAID, vertexIDs[0], readDataA.data());
+    // expected data value received
+    BOOST_TEST(Vector3d(3.1, 3.2, 3.3) == readDataA);
+
+    while (cplInterface.isCouplingOngoing()) {
+      maxDt = cplInterface.advance(maxDt);
+
+      // Scalar Data
+      cplInterface.readBlockScalarData(dataBID, 1, vertexIDs.data(), readDataB.data());
+      // expected data value received
+      BOOST_TEST(5.0 == readDataB[0]);
+      // change value at read destination
+      readDataB[0] = -0.4;
+      BOOST_TEST(-0.4 == readDataB[0]);
+      cplInterface.readScalarData(dataBID, 0, readDataB[0]);
+      // expected data value received
+      BOOST_TEST(5.0 == readDataB[0]);
+
+      // Vector Data
+      cplInterface.readBlockVectorData(dataAID, 1, vertexIDs.data(), readDataA.data());
+      // expected data value received
+      BOOST_TEST(Vector3d(5.1, 5.2, 5.3) == readDataA);
+      // change value at read destination
+      readDataA[0] = -0.4;
+      readDataA[1] = -0.4;
+      readDataA[2] = -0.4;
+      BOOST_TEST(Vector3d(-0.4, -0.4, -0.4) == readDataA);
+      cplInterface.readVectorData(dataAID, vertexIDs[0], readDataA.data());
+      // expected data value received
+      BOOST_TEST(Vector3d(5.1, 5.2, 5.3) == readDataA);
+    }
+    cplInterface.finalize();
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    int             meshTwoID = cplInterface.getMeshID("MeshTwo");
+    int             size      = 1;
+    Eigen::VectorXi vertexIDs(size);
+    Eigen::VectorXd writeDataA(size * 3);
+    Eigen::VectorXd writeDataB(size);
+    Eigen::VectorXd writePositions(size * 3);
+    vertexIDs[0] = cplInterface.setMeshVertex(meshTwoID, writePositions.data());
+
+    double maxDt   = cplInterface.initialize();
+    int    dataAID = cplInterface.getDataID("DataOne", meshTwoID);
+    int    dataBID = cplInterface.getDataID("DataTwo", meshTwoID);
+
+    // Scalar Data
+    writeDataB[0] = -0.1;
+    BOOST_TEST(-0.1 == writeDataB[0]);
+    cplInterface.writeBlockScalarData(dataBID, 1, vertexIDs.data(), writeDataB.data());
+    // expected data value sent, overwriting previous one
+    writeDataB[0] = 3.0;
+    BOOST_TEST(3.0 == writeDataB[0]);
+    cplInterface.writeScalarData(dataBID, vertexIDs[0], writeDataB[0]);
```

Comment:
```suggestion
    // multiple writeBlockScalarData
    writeDataB[0] = -0.1;
    BOOST_TEST(-0.1 == writeDataB[0]);
    cplInterface.writeBlockScalarData(dataBID, 1, vertexIDs.data(), writeDataB.data());
    // expected data value sent, overwriting previous one
    writeDataB[0] = 3.0;
    BOOST_TEST(3.0 == writeDataB[0]);    
    // multiple writeScalarData    
    cplInterface.writeBlockScalarData(dataBID, 1, vertexIDs.data(), writeDataB.data());
```

Test multiple calls to `writeBlockScalarData` here.

Reply:
Imagine that `writeBlockScalarData` is broken and data written via `writeBlockScalarData` is not correctly communicated. This bug might be hidden by calling `writeScalarData` afterwards. (We only see, if something goes wrong here at the other participant)